src/main/java/frc/robot/subsystems/Vision.java

```diff
@@ -441,6 +454,32 @@ private void setExposure(UsbCamera cam, StandardPipeline pipeline) {
       cam.setExposureAuto();
     }
   }
+
+  @Override
+  public void periodic() {
+    double turn = aimAtTarget(); //Get the turn speed from the camera
+    if (turn == previousTurn) {
+      angleError +=  drivetrain.gyroAngle() - currentAngle; 
+      currentAngle = drivetrain.gyroAngle();
+    } else {
+      previousTurn = turn;
+      angleError = turn*PIXELS_TO_DEGREES;
+      currentAngle = drivetrain.gyroAngle();
+    }
+
+    SmartDashboard.putNumber("Angle Error", angleError);
+    turnSpeed = angleError*Drivetrain.AIM_TURN_SPEED;
+
+    if (Math.abs(turnSpeed) < MIN_TURN_SPEED) turnSpeed = Math.copySign(MIN_TURN_SPEED, turnSpeed);
+    if (Math.abs(turnSpeed) > MAX_TURN_SPEED) turnSpeed = Math.copySign(MIN_TURN_SPEED, turnSpeed);
+    if (Math.abs(angleError) > THRESHOLD) {
+      aligned = false;
+    } else {
+      aligned = true;
+    }
+
+    SmartDashboard.putBoolean("ALIGNED", aligned);
+  }
```

Comment:
Why do we need to do this processing constantly in the background?  This is only used when shooting right?  Doesn't make sense to waste CPU cycles when we're not shooting (the rio has a really terrible processor).

Reply:
I can see why you would need to pass the gyro in the do continuous processing.  What I was suggesting is that you only do the calculation when taking a shot in auto or to check the alignment before the drive takes a shot.  Then you can just pass `robot.getGyro()` into the vision calculation function from those command (the command will need to require the drivetrain, but it probably does anyway cause i don't think you would ever be shooting while driving simultaneously.

Reply:
That's a good idea. I changed it so that it only calls the method when the commands need it and the angle gets passed in. I'll test if this still works after school today.